src/scanner.c

```diff
@@ -3085,6 +3087,11 @@ Obj FuncSET_PRINT_FORMATTING_STDOUT(Obj self, Obj val) {
   return val;
 }
 
+Obj FuncGET_FILENAME_CACHE(Obj self)
+{
+  return CopyObj(FilenameCache, 1);
+}
+
```

Comment:
The first three commits of this PR are obviously fine and could go in right away, without waiting for the rest of this PR.